plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -396,6 +403,20 @@ public void checkAndSetLatestUpdateCenter() {
         }
     }
 
+    public String appendPathOntoUrl(String urlString, String pathSection) {
```

Comment:
Is this the best place for this method?
Given it's part of the public API by adding here I would think about either moving it to another class or potentially making it package-private, which will allow unit testing but not make it part of the API

Reply:
Moving it to another class along with the other, similar new method. This also solves the naming issue with the test classes.